docs/specifications/tests/Zone-TP/zone11.md

```diff
@@ -0,0 +1,166 @@
+severity level can be changed in the [Zonemaster-Engine profile]. Also see the
+[Severity Level Definitions] document.
+
+The argument names in the Arguments column lists the arguments used in the
+message. The argument names are defined in the [argument list].
+
+## Test procedure
+
+In this section and unless otherwise specified below, the terms "[DNS Query]"
+follow the specification for DNS queries as specified in [DNS Query and Response Defaults].
+The handling of the DNS responses on the DNS queries follow, unless otherwise specified below,
+what is specified for [DNS Response] in the same specification.
+
+1. Create a [DNS Query] with query type SOA and query name *Child Zone* ("SOA Query")
+
+2. Obtain the set of name server IP addresses using [Method5]
+   ("Name Server IP").
+
+3. Create the following empty sets:
+   1. Name server IP address ("No MNAME Record")
+   2. Name server IP address ("No Serial Record")
+   3. Name server IP address ("MNAME Is Zone Name")
+   4. Name server IP address ("MNAME Not Master")
+   5. Name server IP address ("MNAME Is Master")
+
+4. For each name server IP in *Name Server IP* do:
+   1. Send *SOA Query* to the name server and collect the response.
+   2. Go to next name server IP if at least one of the following criteria is met:
+      1. There is no DNS response.
+      2. [RCODE Name] of the response is not "NoError".
+      3. The AA flag is not set in the response.
+      4. There is no SOA record with owner name matching the query.
+   3. From the DNS response, store the MNAME and its associated SERIAL.
+      1. Go to next name server IP if at least one of the following criteria is met:
+         1. If there is no MNAME record, then add name server IP to the *No MNAME Record* set. 
+         2. If there is no SERIAL record, then add name server IP to the *No SERIAL Record* set. 
+         3. If the name in the MNAME record is the same as *Child Zone*, then add name server IP to the *MNAME Is Zone Name* set.
+   4. Send *SOA Query* to the name server in the MNAME ("MNAME Nameserver").
+      1. If the SOA response has [RCODE Name] "NoError" then:
+         1. If the AA flag is not set, then output *Z11_MNAME_NOT_AUTHORITATIVE* with name server name and IP.
+      2. Else, output *Z11_MNAME_NO_RESPONSE* with name server name and IP and go to next name server IP.
+   5. Obtain the set of name server IP addresses using [Method5] ("NS IP").
+   6. For each name server in *NS IP*, send *SOA Query* to the name server and collect the response.
+      1. If there is a response, compare the SERIAL of that name server to the one from *MNAME Nameserver*.
+         1. If they match, then add name server IP to the *MNAME Is Master* set.
+         2. Else, then add name server IP to the *MNAME Not Master* set.
+   7. Obtain the set of name server IP names using [Method2] ("Glue Names").
+      1. If *MNAME Nameserver* is not part of the *Glue Names* set for the zone, then output *Z11_MNAME_NOT_IN_GLUE*.

```

Comment:
Why not checking against the NS RRset in the zone (method 3) or the combined NS RRset?

Reply:
Fixed